client/my-sites/email/email-providers-stacked-comparison/email-provider-stacked-card/email-provider-generic-form.tsx

```diff
@@ -0,0 +1,868 @@
+			const changedUser = { ...user, [ fieldName ]: { value: fieldValue, error: null } };
+
+			if ( GENERIC_EMAIL_FORM_FIRST_NAME_FIELD === fieldName && ! mailBoxFieldTouched ) {
+				return { ...changedUser, mailBox: { value: sanitizeEmail( fieldValue ), error: null } };
+			}
+
+			return changedUser;
+		} );
+
+		onUsersChange( validateUsers( changedUsers, extraValidation, hiddenFields ) );
+		setValidForm( areAllUsersValid( users, hiddenFields ) );
+	};
+
+	const onUserAdd = () => {
+		onUsersChange( [ ...users, newUser( selectedDomainName ) ] );
+	};
+
+	const onUserRemove = ( uuid: string ) => () => {
+		const newUserList = users.filter( ( _user ) => _user.uuid !== uuid );
+
+		onUsersChange( 0 < newUserList.length ? newUserList : [ newUser( selectedDomainName ) ] );
+	};
+
+	return (
+		<div>
+			{ users.map( ( user, index ) => (
+				<Fragment key={ user.uuid }>
+					<GenericNewUser
+						autoFocus={ autoFocus } // eslint-disable-line jsx-a11y/no-autofocus
+						domains={
+							domains ? domains.map( ( domain: any ) => domain.name ) : [ selectedDomainName ]
+						}
+						hiddenFields={ hiddenFields }
+						user={ user }
+						onUserValueChange={ onUserValueChange( user.uuid ) }
+						onUserRemove={ onUserRemove( user.uuid ) }
+						onReturnKeyPress={ onReturnKeyPress }
+						selectedDomainName={ selectedDomainName }
+						showTrashButton={ index > 0 }
+					/>
+
+					<hr className="email-provider-generic-form__user-divider" />
+				</Fragment>
+			) ) }
+
+			{ showAddAnotherMailboxButton && (
+				<div className="email-provider-generic-form__add-another-user-container">
+					{ /* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/interactive-supports-focus */ }
+					<span
+						role="button"
+						className="email-provider-generic-form__add-another-user-button"
+						onClick={ onUserAdd }
+					>
+						<Gridicon icon="plus" />
+						{ translate( 'Add an extra mailbox' ) }
+					</span>
+					<hr className="email-provider-generic-form__user-divider" />
+				</div>
+			) }
+
+			<div className="email-provider-generic-form__actions">{ children }</div>
+		</div>
+	);
+};
+
+export const transformGenericUserFromTitanMailboxForCart = (
+	genericNewUser: GenericNewUser
+): TitanProductUser => ( {
+	alternative_email: genericNewUser.alternativeEmail?.value,
+	email: `${ genericNewUser.mailBox?.value }@${ genericNewUser.domain?.value }`,
+	is_admin: Boolean( genericNewUser.isAdmin?.value ),
+	name: genericNewUser.fullName?.value ?? genericNewUser.firstName?.value,
+	password: genericNewUser.password?.value,
+} );
+
+const transformGenericUserFromGoogleWorkspaceMailboxForCart = ( {
+	firstName: { value: firstname },
+	lastName: { value: lastname },
+	domain: { value: domain },
+	mailBox: { value: mailBox },
+	password: { value: password },
+}: GSuiteNewUser ): GSuiteProductUser => ( {
+	email: `${ mailBox }@${ domain }`.toLowerCase(),
+	firstname,
+	lastname,
+	password,
+} );
+
+const mapGoogleCartItem = (
```

Comment:
Nit: it might be more appropriate to call this `buildGoogleCartItem()` -- this code isn't actually doing any "mapping".